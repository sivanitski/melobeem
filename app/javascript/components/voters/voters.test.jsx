import { useRequest } from "ahooks";
import { mount } from "enzyme";
import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

import Voters from "./voters";
const mockMatch = {
  params: {
    id: "111",
  },
};

describe("Components", () => {
  describe("Voters", () => {
    it("should render Vote component", () => {
      const component = renderer
        .create(
          <Router>
            <Voters match={mockMatch} />
          </Router>
        )
        .toJSON();

      expect(component).toMatchSnapshot();
    });

    it("should render Loading state", () => {
      beforeEach(() => useRequest.mockReturnValue({ loading: true }));
      const component = mount(
        <Router>
          <Voters match={mockMatch} />
        </Router>
      );

      expect(component.exists(".loading")).toBeTruthy();
    });
  });
});
