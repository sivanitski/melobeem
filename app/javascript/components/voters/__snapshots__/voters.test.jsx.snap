// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Voters should render Vote component 1`] = `
<div
  className="loading"
>
  Loading...
</div>
`;
